### ISSUE TITLE
Memory inefficiencies when building tidy neural dataframe

### ISSUE THREAD

**Issue by dougollerenshaw** [AllenInstitute/mindscope_utilities#18]
A user in the Neuromatch Academy noted that the `build_tidy_cell_df` function uses more memory than is available on google colab when operating on an excitatory experiment with a large cell count. Specifically, this code block to load experiment  877696762, an excitatory experiment with 478 neurons, uses 10-11 GB of memory:

```
import mindscope_utilities
import mindscope_utilities.visual_behavior_ophys as ophys

from allensdk.brain_observatory.behavior.behavior_project_cache import VisualBehaviorOphysProjectCache

data_storage_directory = "/home/dougo/temp" # Note: this path must exist on your local drive
cache = VisualBehaviorOphysProjectCache.from_s3_cache(cache_dir=data_storage_directory)

ophys_experiment_id = 877696762
experiment = cache.get_behavior_ophys_experiment(ophys_experiment_id)

this_experiment_neural_data = ophys.build_tidy_cell_df(experiment)
```

In the colab example that was provided for the course, we added a number of additional metadata columns, which further adds to the memory usage.

**Comment by dougollerenshaw** [comment]
I think the solution is to cast some of the columns to `categorical`. The cell_roi_id and cell_specimen_id columns are obvious candidates. Any additional metadata columns should also be categorical.

**Comment by dougollerenshaw** [comment]
Solved with PR #19
